cpp/src/Slice/Parser.h

```diff
@@ -158,14 +162,61 @@ namespace Slice
         virtual bool shouldVisitIncludedDefinitions() const { return false; }
     };
 
+    // ----------------------------------------------------------------------
+    // GrammarBase
+    // ----------------------------------------------------------------------
+
+    class GrammarBase
+    {
+    public:
+        virtual ~GrammarBase() = default;
+    };
+
+    // ----------------------------------------------------------------------
+    // Metadata
+    // ----------------------------------------------------------------------
+
+    class Metadata final : public virtual GrammarBase
+    {
+    public:
+        Metadata(const std::string& rawMetadata, const std::string& file, int line);
+        std::string_view directive() const;
+        std::string_view arguments() const;
+
+        std::string file() const;
```

Comment:
Originally I did use a `string_view`, but everywhere where we called it was expecting a string.
Eventually wrapping it in a `string` constructor just got cluttered and annoying, so I switched it back.

Reply:
Maybe some future refactoring of all the callers to this function can improve the situation and then it might be more ergonomic to switch it to `string_view`.

Reply:
For accessors, return a `string` (= copy) or a `const string&` (= const reference to field) is fine, and can be preferable to a `string_view`. We prefer `string_view` over ` const string&` for 'in' parameters.